Fixed tilde expansion of module qualified terms

edechter opened the pull request:
Fixes issue #16 

Added a test in t/apply.pl, and other tests pass.

edechter committed:
Fixed tilde expansion of module qualified terms

mndrix commented:
Thanks for the pull request.  It looks like this does the wrong thing for functions that don't have a module qualification.  When [strip_module/3](http://www.swi-prolog.org/pldoc/man?predicate=strip_module/3) is called on a term without a module, it uses the context module.  In the case of macro expansion, the context module is `func`.  So this change causes all functions without a module to be invoked within the `func` module.

You can see an example of the incorrect macro expansion from this pull request in the `tilde` test:

```prolog
?- listing(tilde).
tilde :-
	func:atom_length(foo, A),
	B is A+2,
	B=:=5,
	C is B+3,
	atom_number(D, C),
	D=='8'.
```

It expanded `atom_length(foo,~)` into `func:atom_length(foo,A)`.

Incidentally, this change highlights an omission in the current test suite.  All the functions are apparently builtin so that calling `func:foo(bar)` happens to be the same as calling `foo(bar)`.  It would be nice to have a failing test for that omission along with a fix for module qualification.

Thanks.

edechter committed:
Add test case that fails for issue 16

edechter committed:
Fixed issue 16.

- Use prolog_load_context module for tilde expansion if term being
  expanded is not qualified.
- Added a passing test case.

edechter commented:
Ok, I think I fixed the issues you pointed out.

mndrix commented:
Thanks.  I merged this with a couple touch ups.  It's released as part of v0.4.2
